tests/test_metadata.py

```diff
@@ -0,0 +1,716 @@
+        rowno, contributor_table_data = utils.read_data_from_table(
+            driver, contributor_table_path, check_match=False
+        )
+
+        # Get the user name from the last row which is added recently
+        user = driver.find_element_by_xpath(
+            contributor_table_path + '/tbody/tr[' + str(rowno) + ']/td[2]'
+        )
+        assert new_user in user.text
+
+    @markers.smoke_test
+    @markers.core_functionality
+    def test_edit_resource_information(self, driver, project_metadata_page):
+        """This test verifies that user can add/remove
+        resource information to project metadata."""
+        orig_resource_type = project_metadata_page.resource_type.text
+        orig_resource_language = project_metadata_page.resource_language.text
+
+        WebDriverWait(driver, 5).until(
+            EC.element_to_be_clickable(
+                (By.CSS_SELECTOR, '[data-test-edit-resource-metadata-button]')
+            )
+        ).click()
+
+        # Select 'Book' from the resource type listbox
+        project_metadata_page.resource_type_dropdown.click()
+        WebDriverWait(driver, 5).until(
+            EC.element_to_be_clickable(
+                (
+                    By.CSS_SELECTOR,
+                    '#ember-basic-dropdown-wormhole > div > ul > li>span',
+                )
+            )
+        )
+        project_metadata_page.select_from_dropdown_listbox('Book')
+        # Select 'Benagli' from the resource language listbox
+        project_metadata_page.resource_language_dropdown.click()
+        WebDriverWait(driver, 5).until(
+            EC.element_to_be_clickable(
+                (
+                    By.CSS_SELECTOR,
+                    '#ember-basic-dropdown-wormhole > div > ul > li>span',
+                )
+            )
+        )
+        project_metadata_page.select_from_dropdown_listbox('Bengali')
+
+        project_metadata_page.resource_information_save_button.click()
+        assert orig_resource_type != project_metadata_page.resource_type.text
+        assert orig_resource_language != project_metadata_page.resource_language.text
+
+    @markers.smoke_test
+    @markers.core_functionality
+    def test_edit_support_funding_information(
+        self, session, driver, project_metadata_page, default_project_with_metadata
+    ):
+        """This test verifies that user can add/remove
+        funder information to project metadata."""
+
+        funder_name = 'NFL Charities'
+        # Get the funder information for the given funder name
+        award_title, award_uri, award_number = get_funder_information(funder_name)
+        WebDriverWait(driver, 5).until(
+            EC.element_to_be_clickable(
+                (By.CSS_SELECTOR, '[data-test-edit-funding-metadata-button]')
+            )
+        ).click()
+
+        funder_info = osf_api.get_funder_data_project(
+            session, project_guid=default_project_with_metadata.id
+        )
+        if funder_info is None:
```

Comment:
let's add the `else` statement

`        else:
            WebDriverWait(driver, 5).until(
                EC.element_to_be_clickable(
                    (By.XPATH, '//button[text()="Delete funder"]')
                )
            ).click()
            WebDriverWait(driver, 5).until(
                EC.element_to_be_clickable((By.XPATH, '//button[text()="Add funder"]'))
            ).click()`

Reply:
Done